src/PerpetualPositionRouter.sol

```diff
@@ -167,23 +199,24 @@ contract PerpetualPositionRouter {
   fallback() external payable {
     // The first 11 bytes are padding. The 12th byte will contain the function name which we will
     // use to pick the correct decoding strategy
-    uint8 funcId = uint8(bytes1(msg.data[11:12]));
-    uint168 combinedArgs;
+    uint8 funcId = uint8(bytes1(msg.data[7:8]));
+    uint200 combinedArgs;
     uint256 amount;
     uint256 oppositeAmountBound;
     if (funcId != 5) {
       (combinedArgs, amount, oppositeAmountBound) =
-        abi.decode(msg.data, (uint168, uint256, uint256));
+        abi.decode(msg.data, (uint200, uint256, uint256));
     } else {
-      (combinedArgs, oppositeAmountBound) = abi.decode(msg.data, (uint168, uint256));
+      (combinedArgs, oppositeAmountBound) = abi.decode(msg.data, (uint200, uint256));
     }
     uint160 sqrtPriceLimitX96 = _extractSqrtPriceLimitX96(combinedArgs);
+    uint256 deadline = block.timestamp + (_extractDeadline(combinedArgs) * 1 seconds);
```

Comment:
why do we multiply by `1 seconds`? I think that's a no-op that just multiplies by 1. RR
```suggestion
    uint256 deadline = block.timestamp + _extractDeadline(combinedArgs);
```

Reply:
Good catch, I removed since it behaves as you said

https://github.com/ScopeLift/optimizoors-perpetual-protocol/pull/21/commits/e732a3f8a87d566881b80e64fa7209b4322142b1